sympy/core/add.py

```diff
@@ -625,10 +625,12 @@ def _eval_is_zero(self):
             return None
         b = self.func(*nz)
         if b.is_zero:
-            if not im_or_z and not im:
-                return True
-            if im and not im_or_z:
-                return False
+            if not im_or_z:
+                if im == 0:
+                    return True
+                elif im == 1:
+                    return False
+            return True
```

Comment:
This return True was not there in my suggestion in https://github.com/sympy/sympy/issues/17130#issuecomment-749117670
Why is it needed?

Reply:
I tried doing it without this `return True` line, but then the test I wrote as suggested by you in #17130 was failing.

Since this function was not returning anything when `b.is_zero` is true and `im_or_z` is true, I added this line so as to return true when `b.is_zero` is true and `im_or_z` is true.

Reply:
What test was failing?

The return True here will give erroneous results:
```python
In [2]: Add(1, -1, I, -I, I, evaluate=False).is_zero
Out[2]: True
```

Reply:
It was failing the test you suggested in the [comment](https://github.com/sympy/sympy/issues/17130#issuecomment-749117670). 
It is:
```python
def test_issue_17130():
    e = Add(b, -b, I, -I, evaluate=False)
    assert e._eval_is_zero() == True
```

Reply:
It's fine if that case gives `None`. In the assumptions system we just need to make sure that we don't incorrectly return True or False. There will always be cases that are not resolvable because an assumptions check like `is_zero` needs to be efficient.

The test could just be for None but with a comment indicating why:
```
    assert e._eval_is_zero() is None # ideally this would be True
```

Reply:
I see, thanks for the help. I will change the code accordingly.